streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -282,6 +284,21 @@ public boolean commitNeeded() {
         return Collections.unmodifiableMap(stateMgr.changelogOffsets());
     }
 
+    @Override
+    public Map<TopicPartition, Long> getCommittedOffsets() {
+        return new HashMap<>();
+    }
+
+    @Override
+    public Map<TopicPartition, Long> getHighWaterMark() {
+        return new HashMap<>();
+    }
```

Comment:
Should this two return a `Collections.emptyMap()` instead? Or is it expected that the returned map is modifiable?

Reply:
The maps returned by they different types tasks are intended to modifiable

Reply:
Could you elaborate? Why do we need to modify the results afterwards?